### PR TITLE
Feature/52 Solves #52 

### PR DESCRIPTION
**Pull request by danloa**
* Added a new test file that checks: execution of script task when running correctly, and execution of a script task when it fails
* A new interface ScrtiptTaskInterface was created. It basically has methods to access the script and script format attributes defined by BPMN2
* The activity repository was modified to return new ScriptTasks 
* Added ScriptTask.php file that implements the new ScriptTaskInterface

### CODE REVIEW

File: src/ProcessMaker/Nayra/Contracts/Bpmn/ScriptTaskInterface.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+interface ScriptTaskInterface extends ActivityInterface
+{
+    const BPMN_PROPERTY_SCRIPT_FORMAT = 'scriptFormat';
+    const BPMN_PROPERTY_SCRIPT = 'script';
+
+    /**
+     * Sets the script format of the script task
+     *
+     * @param $scriptFormat
+     * @return mixed
```
Comment (caleeli) [inline src/ProcessMaker/Nayra/Contracts/Bpmn/ScriptTaskInterface.php]: Script format should be a string

File: src/ProcessMaker/Nayra/Contracts/Bpmn/ScriptTaskInterface.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<?php
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Contracts/Bpmn/ScriptTaskInterface.php]: Interface for the script task s

File: tests/Feature/Engine/ScriptTaskTest.php [hunk 3]
```diff
@@ -0,0 +1,216 @@
+<?php
```
Comment (danloa) [inline tests/Feature/Engine/ScriptTaskTest.php]: Tests for the script task element

File: tests/ProcessMaker/Models/ActivityRepository.php [hunk 4]
```diff
@@ -30,6 +30,20 @@ public function createActivityInstance(ProcessInterface $process=null)
         return $activity;
     }
 
+    /**
```
Comment (danloa) [inline tests/ProcessMaker/Models/ActivityRepository.php]: new method of the repository to create script tasks

File: tests/ProcessMaker/Models/ScriptTask.php [hunk 5]
```diff
@@ -0,0 +1,131 @@
+<?php
```
Comment (danloa) [inline tests/ProcessMaker/Models/ScriptTask.php]: Implementation of the ScriptTaskInterface for the tests

File: tests/Feature/Engine/ScriptTaskTest.php [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+            ActivityInterface::EVENT_ACTIVITY_ACTIVATED,
+            ActivityInterface::EVENT_ACTIVITY_COMPLETED,
+            ActivityInterface::EVENT_ACTIVITY_CLOSED,
+            EndEventInterface::EVENT_THROW_TOKEN_ARRIVES,
+            EndEventInterface::EVENT_THROW_TOKEN_CONSUMED,
+            EndEventInterface::EVENT_EVENT_TRIGGERED,
+            ProcessInterface::EVENT_PROCESS_COMPLETED,
+        ]);
+
+        //$this->assertScriptTaskExecuted($scriptTask);
```
Comment (caleeli) [inline tests/Feature/Engine/ScriptTaskTest.php]: remove the commented line

File: tests/Feature/Engine/ScriptTaskTest.php [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+            ActivityInterface::EVENT_ACTIVITY_ACTIVATED,
+            ActivityInterface::EVENT_ACTIVITY_COMPLETED,
+            ActivityInterface::EVENT_ACTIVITY_CLOSED,
+            EndEventInterface::EVENT_THROW_TOKEN_ARRIVES,
+            EndEventInterface::EVENT_THROW_TOKEN_CONSUMED,
+            EndEventInterface::EVENT_EVENT_TRIGGERED,
+            ProcessInterface::EVENT_PROCESS_COMPLETED,
+        ]);
+
+        $this->assertEquals($scriptTask1->getProperty(self::TEST_PROPERTY), 1);
```
Comment (caleeli) [inline tests/Feature/Engine/ScriptTaskTest.php]: please document this assertions

File: tests/Feature/Engine/ScriptTaskTest.php [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+            EventInterface::EVENT_EVENT_TRIGGERED,
+            ActivityInterface::EVENT_ACTIVITY_ACTIVATED,
+        ]);
+
+        //complete the first activity (that is a manual one)
+        $token = $activity1->getTokens($instance)->item(0);
+        $activity1->complete($token);
+        $this->engine->runToNextState();
+
+        $scriptToken = $scriptTask->getTokens($instance)->item(0);
```
Comment (caleeli) [inline tests/Feature/Engine/ScriptTaskTest.php]: Comment a description of this assertion

**Pull request merged**